src/join/leftjoin!.jl

```diff
@@ -0,0 +1,175 @@
+"""
+    leftjoin!(df1, df2; on, makeunique=false, source=nothing,
+              matchmissing=:error)
+
+Perform a left join of two data frame objects by updating the `df1` with the
+joined columns from `df2`.
+
+The rows in `on` columns of `df2` that match rows in `df1` must be unique.
```

Comment:
Yes - we guarantee this. Also the 'uniqueness' restriction makes it possible to perform this operation fast because:
1. we do not touch `df1` columns (so we do no work there)
2. mapping can be done fast (we know upfront we need exactly one slot per row)

I have updated the description in your suggestion above (as it was overlapping).